sdks/python/apache_beam/io/fileio.py

```diff
@@ -34,17 +33,301 @@
+    self.file.close()
+
+  def __iter__(self):
+    if self.current_offset > 0 and ChannelFactory.is_compressed(self.file):
+      # TODO: Turns this warning into an exception soon.
+      logging.warning('Ignoring split starting at (%s) for compressed source.',
+                      self.current_offset)
+      return
+
+    while True:
+      if not self.range_tracker.try_claim(record_start=self.current_offset):
+        # Reader has completed reading the set of records in its range. Note
+        # that the end offset of the range may be smaller than the original
+        # end offset defined when creating the reader due to reader accepting
+        # a dynamic split request from the service.
+        return
+
+      # Note that for compressed sources, delta_offsets are virtual and don't
+      # actually correspond to byte offsets in the underlying file. They
+      # nonetheless correspond to unique virtual position locations.
+      for eof, record, delta_offset in self.read_records():
+        if eof:
+          # Can't read from this source anymore and the record and delta_offset
+          # are non-sensical; hence we are done.
+          return
+        else:
+          self.notify_observers(record, is_encoded=False)
+          self.current_offset += delta_offset
+          yield record
+
+  def seek_to_true_start_offset(self):
+    """Seeks the underlying file to the appropriate start_offset that is
+       compatible with range-tracking and position models and updates
+       self.current_offset accordingly.
+    """
+    raise NotImplementedError
+
+  def read_records(self):
+    """
+      Yields information about (possibly multiple) records corresponding to
+      self.current_offset
+
+      If a read_records() invocation returns multiple results, the first record
+      must be at a split point and other records should not be at split points.
+      The first record is assumed to be at self.current_offset and the caller
+      should use the yielded delta_offsets to update self.current_offset
+      accordingly.
+
+      The yielded value is a tripplet of the form:
+        eof, record, delta_offset
+      eof: A boolean indicating whether eof has been reached, in which case
+        the contents of record and delta_offset cannot be trusted or used.
+      record: The (possibly decoded) record (ie payload) read from the
+        underlying source.
+      delta_offset: The delta_offfset (from self.current_offset) in bytes, that
+        has been consumed from the underlying source, to the starting position
+        of the next record (or EOF if no record exists).
+    """
+    raise NotImplementedError
+
+  def get_progress(self):
+    return iobase.ReaderProgress(position=iobase.ReaderPosition(
+        byte_offset=self.range_tracker.last_record_start))
+
+  def request_dynamic_split(self, dynamic_split_request):
+    if ChannelFactory.is_compressed(self.file):
+      # TODO: Turns this warning into an exception soon.
```

Comment:
Let's add a comment here with something like.

When compression is enabled both initial and dynamic splitting should be prevented. Here we prevent dynamic splitting by ignoring all dynamic split requests at the reader.